src/pool.rs

```diff
@@ -254,10 +242,9 @@ impl ConnectionPool {
     /// Get a connection from the pool.
     pub async fn get(
         &self,
-        shard: usize,           // shard number
-        role: Option<Role>,     // primary or replica
-        process_id: i32,        // client id
-        mut round_robin: usize, // round robin offset
```

Comment:
After randomization, this factor won't be needed. It is only used by the `validate` method to go over all instances. 

To simplify the logic for this method, I refactored `validate` to checkout connection directly from the underlying pool using shard_index and instance_index directly.